### ISSUE TITLE
How many epochs to get EN-FR SOTA result in "Scaling NMT" paper?

### ISSUE THREAD

**Issue by xiaoda99** [pytorch/fairseq#361]
#233 provided a training log of 38 epochs. How many epochs were actually trained to get the EN-FR result reported in Table 2 of "Scaling NMT" paper (43.2 BLEU in 512 min), and how many updates were performed? I have a rough estimate by myself of ~100 epochs ~100k updates. I think it is quite long. Am I correct?
p.s. It would be great of you could share the complete training log.

- Da Xiao

**Comment by myleott** [comment]
The log in #233 is actually the full training log (for one seed): https://github.com/pytorch/fairseq/issues/233#issuecomment-408885881

If you look at the bottom of that log, you can see that we trained for 80k updates, which took 30878 
 seconds ~= 515 minutes.

> I have a rough estimate by myself of ~100 epochs ~100k updates

This is using what batch size?

**Comment by xiaoda99** [comment]
5120 * 8 * 16 = 655k
from last row in Table 1, if 402k batch size is used, updates in 512 min would be 9.7k * 512 / 32 = 155.2k
For 655k batch size, with the assumption that tkn/sec remains almost the same, I estimate updates to be 155.2k * 402k / 655k = 95k

**Comment by myleott** [comment]
Sorry, I'm not sure I understand your logic. If you look at the training log:
- max-tokens is 5120 and the effective batch size (wpb) is 617k
- the average words-per-second (wps) varies per epoch, with a high of around 1.7M and a low of around 1.6M
- in 515 minutes we do 80000 updates (this includes validation and saving checkpoints, which is excluded from the timings in Table 1). This gives an average number of words per second of: (80000 updates * 617k words-per-update)/(515*60 seconds) = 1.6M words-per-second

**Comment by xiaoda99** [comment]
I see. 512 minutes includes validation and saving checkpoints, that may be the reason why the wps is lower than reported in Table 1. Thanks!
